app/response_processor.py

```diff
@@ -9,130 +8,101 @@
 from requests.packages.urllib3.exceptions import MaxRetryError
 
 
+class BadMessageError(Exception):
+    # A bad message is broken in some way that will never be accepted by
+    # the endpoing and as such should be rejected (it will still be logged
+    # and stored so no data is lost)
+    pass
+
+
+class RetryableError(Exception):
+    # A retryable error is apparently transient and may be due to temporary
+    # network issues or misconfiguration, but the message is valid and should
+    # be retried
+    pass
+
+
+class DecryptError(Exception):
+    # Can't even decrypt the message. May be corrupt or keys may be out of step.
+    pass
+
+
 class ResponseProcessor:
 
-    @staticmethod
-    def options():
-        rv = {}
-        try:
-            rv["secret"] = os.getenv("SDX_RECEIPT_RRM_SECRET").encode("ascii")
-        except Exception:
-            # No secret in env
-            pass
-        return rv
-
-    @staticmethod
-    def encrypt(message, secret):
-        """
-        Message may be a string or bytes.
-        Secret key must be 32 url-safe base64-encoded bytes.
-
-        """
-        try:
-            f = Fernet(secret)
-        except ValueError:
-            return None
-        try:
-            token = f.encrypt(message)
-        except TypeError:
-            token = f.encrypt(message.encode("utf-8"))
-        return token
+    def __init__(self, logger):
+        self.logger = logger
+        self.tx_id = ""
+        # self.quarantine_publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_QUARANTINE_QUEUE)
 
-    @staticmethod
-    def decrypt(token, secret):
-        """
-        Secret key must be 32 url-safe base64-encoded bytes or string
+    def process(self, message):
 
-        Returned value is a string.
-        """
+        # Decrypt
         try:
-            f = Fernet(secret)
-        except ValueError:
-            return None
+            message = self._decrypt(token=message, secret=settings.SDX_RECEIPT_RRM_SECRET)
+        except Exception as e:
+            self.logger.exception("Exception decrypting message", exception=e)
```

Comment:
Error log level would be more consistent with other exceptions. Also, either this message should contain a tx_id or could the details of the exception be passed in the raising of DecryptError?